psycop/common/model_training/data_loader/data_loader.py

```diff
@@ -42,16 +42,97 @@ def _check_column_names(self, df: pd.DataFrame):
         else:
             log.debug("No column name checker specified. Skipping column name check.")
 
+    @staticmethod
+    def _check_dataframes_can_be_concatenated(
+        datasets: list[pd.DataFrame],
+        uuid_column: str,
+    ) -> bool:
+        """Check if pred_time_uuid columns are sorted so they can be concatenated
+        instead of joined."""
+        base_uuid = datasets[0][uuid_column]
+        return all(base_uuid.equals(df[uuid_column]) for df in datasets[1:])
+
+    @staticmethod
+    def _check_dataframes_can_be_joined(
+        datasets: list[pd.DataFrame],
+        uuid_column: str,
+    ) -> bool:
+        """Check if pred_time_uuid columns contain the same elements so they can
+        be joined without introducing new rows"""
+        base_uuid = datasets[0][uuid_column]
+        return all(base_uuid.isin(df[uuid_column]).all() for df in datasets[1:])
```

Comment:
I have a very hard time parsing this; we should probably split it out over multiple lines for readability.